### PR TITLE
backup managed cluster to history after deleting it

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Refers: https://issues.redhat.com/browse/ACM-5521

### CODE REVIEW

**Comment by hchenxa** [comment]
@yanmxa , do you think we also need to update the grafana dashboard to select the data from this table history.managed_clusters as well?

**Comment by yanmxa** [comment]
> @yanmxa , do you think we also need to update the grafana dashboard to select the data from this table history.managed_clusters as well?

Yes, I think maybe we need to combine the history and status together for all of the manage_clusters.
```pg
SELECT leaf_hub_name, cluster_name, cluster_id, payload, error
FROM status.managed_clusters
UNION ALL
SELECT leaf_hub_name, cluster_name, cluster_id, payload, error
FROM history.managed_clusters;
```

File: operator/pkg/controllers/hubofhubs/database/3.tables.sql [hunk 1]
```diff
@@ -1,83 +1,110 @@
...
+    CONSTRAINT local_policies_unique_constraint UNIQUE (id, cluster_id, compliance_date)
+);
+
+CREATE TABLE IF NOT EXISTS history.local_compliance_job_log (
+    name varchar(63) NOT NULL,
+    start_at timestamp NOT NULL DEFAULT now(),
+    end_at timestamp NOT NULL DEFAULT now(),
+    total int8,
+    inserted int8,
+    offsets int8,
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/database/3.tables.sql:96]: what is changed comparing the previous one?

File: operator/pkg/controllers/hubofhubs/database/3.tables.sql [hunk 2]
```diff
@@ -293,90 +301,6 @@ CREATE TABLE IF NOT EXISTS event.local_policies (
     CONSTRAINT local_policies_unique_constraint UNIQUE (policy_id, cluster_id, created_at)
 );
 
-ALTER TABLE history.applications DROP CONSTRAINT IF EXISTS applications_pkey;
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/database/3.tables.sql:296]: why remove those?

Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/database/3.tables.sql:296]: This way of writing is the same as the following way of writing, the deletion here just makes it more concise.
```pg
    id uuid PRIMARY KEY,
```

File: operator/pkg/controllers/hubofhubs/database/3.tables.sql (hunk 1 above)
Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/database/3.tables.sql:96]: Nothing changed. I just move the history table together.

**Pull request merged**